docs/building-spark.md

```diff
@@ -49,7 +49,7 @@ To create a Spark distribution like those distributed by the
 to be runnable, use `./dev/make-distribution.sh` in the project root directory. It can be configured
 with Maven profile settings and so on like the direct Maven build. Example:
 
-    ./dev/make-distribution.sh --name custom-spark --pip --r --tgz -Psparkr -Phadoop-2.7 -Phive -Phive-thriftserver -Pmesos -Pyarn
+    ./dev/make-distribution.sh --name custom-spark --pip --r --tgz -Psparkr -Phadoop-2.7 -Phive -Phive-thriftserver -Pmesos -Pyarn -Pkubernetes
```

Comment:
should we use k8s? I kept bringing this up and that's because I can never spell Kubernetes properly.

Reply:
It's `k8s://` in the URL scheme we use and the package names are also `k8s` - so, users should never have to type the name.
This is one of the last few holdouts. I'd say that it's consistent here, with the use of other cluster manager names in full in their maven projects. I can change it here if you feel strongly about it.

Reply:
I actually second @rxin, I do get the spelling wrong at times :-)

Reply:
Are you both also referring to the config options etc, which are still `spark.kubernetes.*`, or just the maven build target? If it's everything, it would be a fairly large change.

Reply:
Ping @mridulm, @rxin - can you please confirm the scope of the renaming you were referring to here? Is it just the maven target? Changing all the config options etc would be a considerably large change at this point. Also, a point that was brought up today was - while k8s is common shorthand, it's not universal as is the full name.

Reply:
I've filed https://issues.apache.org/jira/browse/SPARK-22853 to discuss this and unblock this PR. We should be able to reach consensus by release time. :)

Reply:
Yea I don't think you need to block this pr with this.